### ISSUE TITLE
[BUG] getting the loss value as `0.00` from model.evaluate() for Two-Tower retrieval model

### ISSUE THREAD

**Issue by rnyak** [NVIDIA-Merlin/models#291]
### Bug description
When I perform the `model.evaluate` step after training a TT model. I get the following metrics printed out.

`model.evaluate(valid, return_dict=True, batch_size=1024)`
```
{'recall_at_10': 2.1537676730076782e-05,
 'ndcg_10': 9.275297088606749e-06,
 'loss': 0.0,
 'regularization_loss': 0.0,
 'total_loss': 0.0}
```

### Steps/Code to reproduce bug

Please run the Two-Tower retrieval notebook [here](https://github.com/NVIDIA-Merlin/models/blob/retrieval/examples/04-Retrieval-Model.ipynb) to reproduce.

### Expected behavior
Given ndcg@10 value,  Loss value cannot be 0.00. This seems off.

### Environment details
- Merlin version: 
- Platform:
- Python version:
- PyTorch version (GPU?):
- Tensorflow version (GPU?):

using `nvcr.io/nvidia/merlin/merlin-tensorflow-training:22.03` docker image with the latest main branch pulled.
